src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -151,27 +125,52 @@ public static SchemaIndexDef ReadFrom(ByteStringContext context, byte* location,
                 indexDef.IsGlobal = Convert.ToBoolean(*currentPtr);
 
                 currentPtr = input.Read(4, out currentSize);
-                indexDef.NameAsSlice = Slice.From(context, currentPtr, currentSize, ByteStringType.Immutable);
-
-                currentPtr = input.Read(5, out currentSize);
-                var hasName = Convert.ToBoolean(*currentPtr);
-                if (hasName)
-                {
-                    currentPtr = input.Read(6, out currentSize);
-                    indexDef.Name = Encoding.UTF8.GetString(currentPtr, currentSize);
-                }
+                indexDef.Name = Slice.From(context, currentPtr, currentSize, ByteStringType.Immutable);
 
                 return indexDef;
             }
+
+            public void Validate(SchemaIndexDef actual)
+            {
+                if (actual == null)
+                    throw new ArgumentNullException(nameof(actual), "Expected an index but received null");
+
+                if (!SliceComparer.Equals(Name, actual.Name))
+                    throw new ArgumentException(
+                        string.Format("Expected index to have Name='{0}', got Name='{1}' instead", Name, actual.Name),
+                        nameof(actual));
+
+                if (Type != actual.Type)
+                    throw new ArgumentException(
+                        string.Format("Expected index {0} to have Type='{1}', got Type='{2}' instead", actual.Name, Type,
+                            actual.Type),
+                        nameof(actual));
+
+                if (StartIndex != actual.StartIndex)
+                    throw new ArgumentException(
+                        string.Format("Expected index {0} to have StartIndex='{1}', got StartIndex='{2}' instead",
+                            actual.Name, StartIndex, actual.StartIndex),
+                        nameof(actual));
+
+                if (Count != actual.Count)
+                    throw new ArgumentException(
+                        string.Format("Expected index {0} to have Count='{1}', got Count='{2}' instead", actual.Name,
+                            Count, actual.Count),
+                        nameof(actual));
+
+                if (IsGlobal != actual.IsGlobal)
+                    throw new ArgumentException(
+                        string.Format("Expected index {0} to have IsGlobal='{1}', got IsGlobal='{2}' instead",
+                            actual.Name, IsGlobal, actual.IsGlobal),
+                        nameof(actual));
+            }
         }
```

Comment:
Use string interpolation for better string handling. 
https://msdn.microsoft.com/en-us/magazine/dn879355.aspx